src/Room/Message.tsx

```diff
@@ -108,81 +114,123 @@ const MessageHeader: React.FC<{ message: MessageType }> = ({ message }) => {
   const playedAt = message.roomPlaylistRecord?.playedAt && moment(message.roomPlaylistRecord?.playedAt)
   const user = useUserContext()
   const showPin = user.id === message.user.id && !!message.song
+  console.log(message.pinned)
 
   return (
-    <Flex
-      alignItems="flex-start"
-      justifyContent="space-between"
-      sx={{
-        position: 'relative',
-        overflow: 'visible',
-      }}
-    >
-      <Text
+    <>
+      <Box
+        className="message-options"
         sx={{
-          alignItems: 'flex-start',
+          alignItems: 'center',
+          border: '1px solid',
+          borderColor: 'accent',
+          bg: 'background',
+          borderRadius: 6,
+          boxShadow: 'xxl',
+          display: 'none',
           justifyContent: 'space-between',
-          flexDirection: 'column',
-          display: 'flex',
-          fontSize: 2,
-          mb: 2,
-          overflow: 'visible',
-          width: '100%',
+          position: 'absolute',
+          top: -3,
+          right: 4,
         }}
       >
-        <Box mb={1}>
-          <Box as="span" sx={{ color: 'text', fontWeight: '800' }}>
-            {message.user.name}
-          </Box>
+        <Pin messageId={message.id} previouslyPinned={message.pinned} showPin={showPin} />
+      </Box>
 
-          <Box as="span" sx={{ color: 'gray500', fontSize: 1, fontWeight: '600', px: 2 }}>
-            {createdAt.format('ddd h:mm a')}
-          </Box>
+      <Flex
+        alignItems="flex-start"
+        justifyContent="space-between"
+        sx={{
+          position: 'relative',
+          overflow: 'visible',
+        }}
+      >
+        <Text
+          sx={{
+            alignItems: 'center',
+            display: 'flex',
+            fontSize: 2,
+            mb: 1,
+            overflow: 'visible',
+            width: '100%',
+          }}
+        >
+          <Box sx={{ color: 'text', fontWeight: '800' }}>{message.user.name}</Box>
 
+          <Box sx={{ color: 'gray500', fontSize: 1, fontWeight: '600', px: 2 }}>{createdAt.format('ddd h:mm a')}</Box>
           <PlayedAt messageCreated={createdAt} playedAt={playedAt} song={message.song} />
-        </Box>
-      </Text>
-
-      <Pin messageId={message.id} previouslyPinned={message.pinned} showPin={showPin} />
-    </Flex>
+        </Text>
+      </Flex>
+    </>
   )
 }
 
 const Message: React.FC<{ message: MessageType }> = ({ message }) => {
+  const PinnedMessage: React.FC = () => {
```

Comment:
Nit:  Pull this out of `Message` just above like:
```
const PinnedMessage: React.FC<{ pinned: boolean }> = ({ pinned }) => {
  if (!pinned) {
    return <></>
  }
  ...the rest of the flex stuff
}
```

Remove the ternary down below and call with `<PinnedMessage pinned={message.pinned} />`